modules/benchmarks_db/src/lib.rs

```diff
@@ -0,0 +1,529 @@
+    let new_hash = calculate_hash(&targetable.quad) as i64;
+    targetable.quad = new_hash;
+    GameTargetableState::update_by_entity_id(&entity_id, targetable);
+
+    // If the entity is alive (which it should be),
+    // also update the `LiveTargetableState` used by `enemy_ai_agent_loop`.
+    if GameLiveTargetableState::filter_by_entity_id(&entity_id).is_some() {
+        GameLiveTargetableState::update_by_entity_id(
+            &entity_id,
+            GameLiveTargetableState {
+                entity_id,
+                quad: new_hash,
+            },
+        );
+    }
+    let mobile_entity =
+        GameMobileEntityState::filter_by_entity_id(&entity_id).expect("GameMobileEntityState Entity ID not found");
+    let mobile_entity = GameMobileEntityState {
+        entity_id,
+        location_x: mobile_entity.location_x + 1,
+        location_y: mobile_entity.location_y + 1,
+        timestamp: moment_milliseconds(),
+    };
+
+    GameEnemyAiAgentState::update_by_entity_id(&entity_id, agent.clone());
+
+    GameMobileEntityState::update_by_entity_id(&entity_id, mobile_entity);
+}
+
+fn agent_loop(
+    mut agent: GameEnemyAiAgentState,
+    agent_targetable: GameTargetableState,
+    surrounding_agents: &[GameTargetableState],
+    current_time_ms: u64,
+) {
+    let entity_id = agent.entity_id;
+    let coordinates =
+        GameMobileEntityState::filter_by_entity_id(&entity_id).expect("GameMobileEntityState Entity ID not found");
+
+    let agent_entity = GameEnemyState::filter_by_entity_id(&entity_id).expect("GameEnemyState Entity ID not found");
+    let agent_herd = GameHerdCache::filter_by_id(&agent_entity.herd_id).expect("GameHerdCache Entity ID not found");
+    let agent_herd_coordinates = agent_herd.location;
+
+    move_agent(&mut agent, agent_herd_coordinates, current_time_ms);
+}
+
+// We check only for a single pass in the game loop.
+#[spacetimedb(reducer)]
+pub fn game_loop_enemy_ia(players: u64) {
+    let mut count = 0;
+    let current_time_ms = moment_milliseconds();
+
+    for mut agent in GameEnemyAiAgentState::iter() {
+        if agent.next_action_timestamp > current_time_ms {
+            continue;
+        }
+
+        let agent_targetable = GameTargetableState::filter_by_entity_id(&agent.entity_id)
+            .expect("No TargetableState for AgentState entity");
+
+        let surrounding_agents = get_targetables_near_quad(agent_targetable.entity_id, players);
+
+        agent.action = AgentAction::Fighting;
+
+        agent_loop(agent, agent_targetable, &surrounding_agents, current_time_ms);
+
+        count += 1;
+    }
+
+    println!("ENEMY IA LOOP PLAYERS: {players}, processed: {count}");
+}
+
+#[spacetimedb(reducer)]
+pub fn game_benchmark(initial_load: u8) {
+    let initial_load = initial_load as u32;
+    let small_table = initial_load * 10;
+    let num_players = (initial_load * 100) as u64;
+    let big_table = initial_load * 10_000;
+    let biggest_table = initial_load * 20_000;
```

Comment:
These numbers mean that the main thing we're benchmarking here are operations on big_table and biggest_table.

Do we know what the sizes of the corresponding tables look like in bitcraft?